### PR TITLE
Invite organization email and assign points if signed in.

### LINKED ISSUE #4550: Have a way for users to invite organizations they would like to work with it will generate an email they can send mentioning how BLT works and what benefits it brings to their organization

### PR DESCRIPTION
**Pull request by sidd190**
# Organization Invite Points System Implementation

Fixes #4550 

## Summary

Successfully implemented the points awarding logic for organization invites. When a user invites an organization and that organization registers via the referral link, the inviting user receives 5 points.

**Current Invite functionality**

https://github.com/user-attachments/assets/891a989c-2280-46e5-8aa2-ac0ac6738fee

**Implemented invite functionality**

[Screencast from 2025-11-01 00-05-42.webm](https://github.com/user-attachments/assets/f1aabb75-a926-4507-a457-39b2daa3f70f)

### 1. Fixed Organization Registration View (`website/views/company.py`)
- **Added imports**: `InviteOrganization`, `Points`
- **Added points logic**: After successful organization creation, check for referral code in session
- **Award 5 points**: Create Points record for the inviting user
- **Update invite record**: Mark `points_awarded=True` and link to created organization
- **Prevent double points**: Only award points if `points_awarded=False`

### 2. Improved Invite View (`website/views/core.py`)
- **Fixed placeholder issue**: Removed creation of fake placeholder invites
- **Generate sample links**: Use UUID for display-only referral links
- **Better context**: Added proper template variables for email generation
- **Security improvement**: Sample links can't be used to claim points

### 3. Updated Template (`website/templates/invite.html`)
- **Sample link indicator**: Different messages for real vs sample referral links
- **User context**: Show appropriate messages based on login status

### 4. Consolidated URL Routing (`blt/urls.py`)
- **Single invite view**: Use `invite_organization` function from core.py
- **Removed duplicate**: Eliminated unused `InviteCreate` class from user.py

### 5. Comprehensive Tests (`website/test_organization_invite.py`)
- **Points awarding**: Test that 5 points are awarded on successful registration
- **No double points**: Test that same referral code can't be used twice
- **Invalid codes**: Test that invalid referral codes don't award points
- **Sample links**: Test that sample links don't award points
- **Anonymous users**: Test behavior for non-logged-in users

## How It Works

1. **User creates invite**: Logged-in user fills invite form → `InviteOrganization` record created
2. **Referral link generated**: Unique referral code appended to organization registration URL
3. **Organization visits link**: Referral code stored in session
4. **Organization registers**: Registration form submitted successfully
5. **Points awarded**: System finds matching invite record and awards 5 points to sender
6. **Invite updated**: Mark as used and link to created organization

## Key Features

**5 points per successful invite**  
**Prevents duplicate point awards**  
**Secure referral tracking**  
**Works for logged-in users only**  
**Sample links for demonstration**  
**Comprehensive test coverage**  

## Security Considerations

- Only real invite records (not sample links) can award points
- Referral codes are unique UUIDs
- Points only awarded once per invite
- Session-based referral tracking prevents tampering
- Invalid referral codes are handled gracefully

## Testing

Run the test suite to verify functionality:
```bash
python manage.py test website.test_organization_invite
```

The implementation includes tests for all major scenarios including edge cases and security considerations.

## Summary by CodeRabbit

* **New Features**
  * Organization invite flow: generate professional, customizable invitation emails with preview, per-field copy, "Copy All" and "Open in Email Client", plus referral links and referral-based points when an invited organization registers.

* **Bug Fixes / UX**
  * Improved form messaging, field-specific error reporting, conditional copy for logged-in vs guest users, robust referral/session handling and safer URL parsing.

* **Tests**
  * End-to-end coverage for invite workflow, referral handling, and point-awarding logic.

* **Documentation**
  * Expanded invite docs covering workflow, email content, accessibility, and chatbot guidance.

### CODE REVIEW

File: website/views/core.py [hunk 1]
```diff
@@ -2896,3 +2897,117 @@ def get_context_data(self, **kwargs):
...
+            # Create invite record for logged-in users
+            invite_record = InviteOrganization.objects.create(
+                sender=request.user, email=email, organization_name=organization_name
+            )
+
+            # Generate referral link
+            base_url = request.build_absolute_uri(reverse("register_organization"))
+            default_referral = request.build_absolute_uri(reverse("register_organization"))
+            referral_link = context.get("referral_link", default_referral)
+            referral_link = f"{base_url}?ref={invite_record.referral_code}"
```
[the remark this answers is not in the document]
Comment (sidd190) [reply website/views/core.py]: Can i remove  referral_link = f"{base_url}?ref={invite_record.referral_code}"
instead of the default referral logic?

File: INVITE_ORGANIZATION_FEATURE.md
Comment (DonnieBLT) [inline INVITE_ORGANIZATION_FEATURE.md]: remove this

File: .gitignore [hunk 2]
```diff
@@ -15,6 +15,7 @@ geckodriver.log
 geckodriver
 chromedriver
 requirements.txt
+cookies.txt
```
Comment (DonnieBLT) [inline .gitignore]: remove this

File: website/templates/leaderboard_global.html [hunk 3]
```diff
@@ -120,28 +130,32 @@ <h1 class="page-header">Global Leaderboard</h1>
...
+                                        {% if leader.reviewer__user__username %}
+                                            <img src="https://github.com/{{ leader.reviewer__user__username }}.png"
                                                  class=" size-11 select-none border border-gray-200 rounded-full"
-                                                 alt="{{ leader.reviews__reviewer__user__username }}"
+                                                 alt="{{ leader.reviewer__user__username }}"
                                                  width="50px"
                                                  height="50px">
                                         {% else %}
-                                            <img src="{% gravatar_url leader.reviews__reviewer__user__email 50 %}"
+                                            <img src="{% gravatar_url leader.reviewer__user__email 50 %}"
```
Comment (DonnieBLT) [inline website/templates/leaderboard_global.html]: don't mix other issues please

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
some changes requested, see comments

File: website/templates/leaderboard_global.html (hunk 3 above)
Comment (sidd190) [reply website/templates/leaderboard_global.html]: Reverted changes.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

File: test_invite.py
[the remark this answers is not in the document]
Comment (sidd190) [reply test_invite.py]: Yes I was using it initially locally to double check the functionality and added by mistake.

File: website/templates/invite.html [hunk 4]
```diff
@@ -5,66 +5,176 @@
...
+                            </p>
+                        {% else %}
+                            <p>
+                                {% trans "Invite an organization to join BLT." %} <a href="{% url 'account_login' %}"
+    class="font-bold text-[#e74c3c] hover:underline">{% trans "Login" %}</a> {% trans "to earn +5 points when they register via your referral link." %}
+                            </p>
+                        {% endif %}
                         <p>
-                            {% trans "Invite an organization to join and earn" %} <span class="font-bold text-red-500/70">{% trans "+5 points" %}</span> {% trans "when they do." %}
+                            {% trans "Generate a professional invitation email with detailed information about BLT's benefits for organizations." %}
```
[the remark this answers is not in the document]
Comment (sidd190) [reply website/templates/invite.html]: Ahh, yeah makes sense. I'll update it.
